grails-app/services/org/pih/warehouse/shipping/ShipmentService.groovy

```diff
@@ -2155,4 +2156,28 @@ class ShipmentService {
             throw new ValidationException("Invalid shipment", shipment.errors)
         }
     }
+
+    Container initializeContainerFromShipment(Shipment shipment){
+        Container container = new Container()
+        container.shipment = shipment
```

Comment:
I think Grails is expecting the following line instead of / in addition to this one.
```
shipment.addToContainers(container)
```

Reply:
Except we're discarding so maybe container.shipment is what you need to get the JSON to work?